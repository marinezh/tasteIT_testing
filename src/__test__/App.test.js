import { render, screen } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/Browse Recipes/i);
  expect(linkElement).toBeInTheDocument();
});

test("renders learn react link2", () => {
  render(<App />);
  const linkElement = screen.getByText(/BC Helsinki hompepage/i);
  expect(linkElement).toBeInTheDocument();
});

test("first snapshot test", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
