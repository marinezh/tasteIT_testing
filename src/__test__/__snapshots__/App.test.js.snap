// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot test 1`] = `
<div
  className="layout"
>
  <header
    className="header"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <div
        className="logo_wrap"
      >
        <span
          className="material-symbols-outlined"
        >
          restaurant
        </span>
        <h2
          className="logo"
        >
          TasteIT App
        </h2>
      </div>
    </a>
    <nav
      className="nav"
    >
      <ul
        className="ul"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className=""
            href="/allrecipes"
            onClick={[Function]}
          >
            All Recipes
          </a>
        </li>
        <li>
          <a
            className=""
            href="/addrecipe"
            onClick={[Function]}
          >
            Add New Recipe
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <div
      className="home"
    >
      <div>
        <div>
          <video
            autoPlay={true}
            loop={true}
            muted={true}
            src="video1.mp4"
          />
        </div>
        <div
          className="hero_content"
        >
          <h3
            className="hero_title"
          >
            Application where people all around the world can share their favourite recipes.
          </h3>
          <p
            className="hero_title"
          >
            Made in REACT22S Group with Margit Tennosaar
          </p>
        </div>
      </div>
      <div
        className="menu"
      >
        <div
          className="menu_card"
        >
          <h4>
            Browse Recipes
          </h4>
          <p>
            Search for your favorite recipe
          </p>
          <a
            className=""
            href="/allrecipes"
            onClick={[Function]}
          >
            All recipes
          </a>
        </div>
        <div
          className="menu_card"
        >
          <h4>
            Add new recipes
          </h4>
          <p>
            Add new recipe from your own country
          </p>
          <a
            className=""
            href="/addrecipe"
            onClick={[Function]}
          >
            Add new recipe
          </a>
        </div>
        <div
          className="menu_card"
        >
          <h4>
            Learn more about our projects
          </h4>
          <p>
            Visit our programme homepage
          </p>
          <a
            href="https://en.bc.fi/qualifications/full-stack-web-developer-program/"
            rel="noreferrer"
            target="_blank"
          >
            BC Helsinki hompepage
          </a>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <p
      className="footer_info"
    >
      Marina Zhivotova Â© 2023
    </p>
  </footer>
</div>
`;
